core/src/main/scala/org/apache/spark/rpc/RpcEnv.scala

```diff
@@ -182,3 +185,130 @@ private[spark] object RpcAddress {
     RpcAddress(host, port)
   }
 }
+
+
+/**
+ * An exception thrown if RpcTimeout modifies a [[TimeoutException]].
+ */
+private[rpc] class RpcTimeoutException(message: String)
+  extends TimeoutException(message)
```

Comment:
I just realized that `TimeoutException` doesn't allow you to specify the original cause.  This is a little unfortunate, because it means all the stack traces will end where we create this class.  It looks like there is an `initCause` method which does what we want though.  So I think the constructor should require a `cause: TimeoutException`, which we set via `initCause`.